harness-warmup/harness.rb

```diff
@@ -0,0 +1,72 @@
+require 'benchmark'
+require_relative '../harness/harness-common'
+require_relative '../misc/stats'
+
+MIN_ITERS = Integer(ENV['MIN_ITERS'] || 10)
+MIN_TIME = Integer(ENV['MIN_TIME'] || 5)
+MAX_TIME = Integer(ENV['MAX_TIME'] || 20 * 60)
```

Comment:
With the caveat that any threshold has some amount of arbitrariness, I would reduce the default `MAX_TIME` to something like 10 minutes. 20 seems overly generous 🤷‍♀️

Reply:
MJIT did hit the 20min timeout in https://gist.github.com/k0kubun/10ae929640aabf933fb93f2059b9fa35#file-ruby-3-1-3-mjit-log and it wasn't fully warmed up yet.
TruffleRuby JVM CE took 7min in that run.
FWIW I computed that with:
```ruby
lines=File.readlines("ruby-3.1.3-mjit.log")
p lines.grep(/iter.+(\d+)ms/).map { |l| Integer l[/(\d+)ms/, 1] }.sum/1000.0/60
```

So I think 20 minutes is safer. It's meant to be a "avoid running forever in case it's too unstable" not a limit on how much warmup should be allowed which seems problematic because that depends on the machine (e.g. on a slower machine TruffleRuby could need more than 10min on the bigger benchmarks), etc.

That makes me think it would be nice to print how long all iterations took (I have it in the PR description via `REPORTTIME=3` bu that's zsh-specific). I can add that to the output.